tests/components/wmspro/test_diagnostics.py

```diff
@@ -0,0 +1,38 @@
+"""Test the wmspro diagnostics."""
+
+from unittest.mock import AsyncMock
+
+from homeassistant.core import HomeAssistant
+
+from . import setup_config_entry
+
+from tests.common import MockConfigEntry
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+from tests.typing import ClientSessionGenerator
+
+
+async def test_diagnostics(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    mock_config_entry: MockConfigEntry,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_test: AsyncMock,
+    mock_dest_refresh: AsyncMock,
+) -> None:
+    """Test that a config entry can be loaded with DeviceConfig."""
+    assert await setup_config_entry(hass, mock_config_entry)
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_test.mock_calls) == 1
+    assert len(mock_dest_refresh.mock_calls) == 2
+
+    result = await get_diagnostics_for_config_entry(
+        hass, hass_client, mock_config_entry
+    )
+    assert result["control"]["_control"] == "http://webcontrol/commonCommand"
+    assert result["control"]["_host"] == "webcontrol"
+    assert len(result["control"]["dests"]) == len(result["dests"])
+    assert len(result["control"]["rooms"]) == len(result["rooms"])
+    assert len(result["control"]["scenes"]) == len(result["scenes"])
+    assert len(result["dests"]) == 2
+    assert len(result["rooms"]) == 1
+    assert len(result["scenes"]) == 1
```

Comment:
Please use snapshot tests for this

Reply:
I tried this, but since the serialized data contains dynamic objects that is not possible.